python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -160,9 +160,28 @@ def test_integer_index_column(self):
         df = pd.DataFrame([(1, 'a'), (2, 'b'), (3, 'c')])
         _check_pandas_roundtrip(df, preserve_index=True)
 
+    def test_index_metadata_field_name(self):
+        df = pd.DataFrame(
+            [(1, 'a'), (2, 'b'), (3, 'c')],
+            index=pd.MultiIndex.from_arrays(
+                [['c', 'b', 'a'], [3, 2, 1]],
+                names=[None, 'foo']
+            )
+        )
+        t = pa.Table.from_pandas(df, preserve_index=True)
+        raw_metadata = t.schema.metadata
+
+        js = json.loads(raw_metadata[b'pandas'].decode('utf8'))
+
+        _, _, idx0, foo = js['columns']
+        idx0_name, foo_name = js['index_columns']
+        assert idx0_name == '__index_level_0__'
+        assert idx0['field_name'] == idx0_name
+
```

Comment:
Should we also assert that `idx0['name']` is None, or is that already tested elsewhere adequately?

Reply:
I'll add that